fda/functional_data.py

```diff
@@ -1011,4 +1011,15 @@ def __truediv__(self, other):
     def __rtruediv__(self, other):
         """Right division for FData object."""
 
-        pass
\ No newline at end of file
+        pass
+
+    def __iter__(self):
```

Comment:
I though that we already had the iteration, as we implement `__getitem__`. Is it not true?

Reply:
Yes, it could be iterated, but the result is an infinite loop.
```python
for sample in fd: # fd without __iter__
   do_something(sample) # Infinite loop
```
And the objects cannot be unzipped.

Without `__len__` the iteration works well, but I like the syntax `len(fd)`.